Tensile/Source/lib/source/ContractionSolution.cpp

```diff
@@ -419,7 +419,8 @@ namespace Tensile
             }
         }
 
-        if(sizeMapping.globalAccumulation && sizeMapping.streamK < 2)
+        if(sizeMapping.globalAccumulation
```

Comment:
This part looks inconsistent.
In SolutionStruct.py, _GlobalAccumulation is set only if streamKAtomic = 0.
However, the if condition here is (globalAccumulation and streamKAtomic).

I am thinking we just need to check globalAccumulation here.

Reply:
I reviewed the code, and I think it is right the way it is now. For non-atomic stream-k, we set GlobalAccumulation to PartialsBuffer, which is handled separated from the ws_d cause size isn't dependent on D dimensions. For atomic stream-k, GlobalAccumulation is not set because WS is not needed

Reply:
I am a little confused.
This condition is false if streamKAtomic=1 because globalAccumulation is not set, right?
Should this be like this?
  (globalAccumulation && streamK==0) || streamKAtomic

Reply:
Oh, yes you're right, that makes the condition false. For stream-k atomic case, workspace pointers don't actually get set in the contraction inputs, and it should skip these arguments. I've updated it to just check
(globalAccumulation && streamK == 0)
Reran the stream-k atomic test locally to make sure it still passes.